### ISSUE TITLE
Static assets not working in ASP.NET Core MVC Views

### ISSUE THREAD

**Issue by jmaguilar** [dotnet/aspnetcore#59003]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

The static assets feature does not seem to work with ASP.NET Core MVC views.

Despite configuring the application properly to use static assets, the rendered page code always references the original source file names without fingerprints and the import map is not generated.

### Expected Behavior

After the MVC View is rendered, the generated page should:

- Include the import map in the `<head>` section.
- Use fingerprinted resource names for static file references.

### Steps To Reproduce

Create a new ASP.NET Core MVC project and launch it.

### .NET Version

9.0.100

### Anything else?

First, note that **this is only happening in MVC Views**. If we use Razor Pages with static assets, the rendered pages will be using the fingerprinted resources and the import map is generated fine.

Additionally, I've been digging a bit, and I think that the problem is that `MapControllerRoute()` -and other similar methods like `MapDefaultControllerRoute()`- are returning a builder that does not contain the item required by `WithStaticAssets()` to do its magic.

In fact, we can make it work using a bit of reflection: 

```cs
... // Other Program.cs code

// Replace this:
// app.MapControllerRoute(
//    name: "default",
//    pattern: "{controller=Home}/{action=Index}/{id?}")
//    .WithStaticAssets();

// For this:
var controllerActionEndpointConventionBuilder = app.MapControllerRoute(
    name: "default",
    pattern: "{controller=Home}/{action=Index}/{id?}");

var itemsProperty = controllerActionEndpointConventionBuilder
                                    .GetType()
                                    .GetProperty("Items", BindingFlags.NonPublic | BindingFlags.Instance);

var itemsPropertyDictionary = itemsProperty.GetValue(controllerActionEndpointConventionBuilder) as Dictionary<string, object>;
itemsPropertyDictionary?.Add("__EndpointRouteBuilder", app);
controllerActionEndpointConventionBuilder.WithStaticAssets();
// End patch
```

I hope it can help to pinpoint the root cause of the problem.

**Comment by dinowilliam** [comment]
Yes, it's a problem in 9. I have downgraded it to 8, and it is working. I don't even need the static assets method.

**Comment by DavidZidar** [comment]
The issue is that `MapControllerRoute()` and `MapDefaultControllerRoute()` call `.AddRoute()` on their respective `ControllerActionEndpointDataSource`-instances and returning the result from that method call which results in an instance of `ControllerActionEndpointConventionBuilder` that does not have endpoint builder in the `Items` collection that `.WithStaticAssets()` require, in fact the `Items` collection is empty.

I'd say this is both a bug in the respective methods and also a bug in the `dotnet new mvc` template which sets up the project in a non working state.

I have gotten static assets to work properly in MVC views simply by doing the following:
```csharp
app.MapControllers()
	.WithStaticAssets();

app.MapDefaultControllerRoute();
```

`.MapControllers()` do not call `.AddRoute()` so it does have the endpoint builder in the `Items` collection and suddenly everything works.

**Comment by DavidZidar** [comment]
I just created issue #59152 to increase visibility of this issue and to point out that the `dotnet new mvc` template isn't properly functional.
